ios/tunnel/untrusted.go

```
@@ -0,0 +1,483 @@
+	proofTlv.writeData(typePublicKey, srp.ClientPublic)
+	proofTlv.writeData(typeProof, srp.ClientProof)
+
+	err = t.controlChannel.writeEvent(&pairingData{
+		data: proofTlv.bytes(),
+		kind: "setupManualPairing",
+	})
+	if err != nil {
+		return nil, fmt.Errorf("setupSessionKey: failed to send SRP proof: %w", err)
+	}
+
+	var proofPairingData pairingData
+	err = t.controlChannel.readEvent(&proofPairingData)
+	if err != nil {
+		return nil, fmt.Errorf("setupSessionKey: failed to read device SRP proof: %w", err)
+	}
+
+	serverProof, err := tlvReader(proofPairingData.data).readCoalesced(typeProof)
+	if err != nil {
+		return nil, fmt.Errorf("setupSessionKey: failed to parse device proof: %w", err)
+	}
+	verified := srp.verifyServerProof(serverProof)
+	if !verified {
+		return nil, fmt.Errorf("setupSessionKey: could not verify server proof")
+	}
+	return srp.SessionKey, nil
+}
+
+func (t *tunnelService) exchangeDeviceInfo(sessionKey []byte) error {
+	hkdfPairSetup := hkdf.New(sha512.New, sessionKey, []byte("Pair-Setup-Controller-Sign-Salt"), []byte("Pair-Setup-Controller-Sign-Info"))
+	buf := bytes.NewBuffer(nil)
+	io.CopyN(buf, hkdfPairSetup, 32)
+	buf.WriteString(t.pairRecords.selfId.Identifier)
+	buf.Write(t.pairRecords.selfId.publicKey())
+
+	signature := ed25519.Sign(t.pairRecords.selfId.privateKey(), buf.Bytes())
+
+	deviceInfo, err := opack.Encode(map[string]interface{}{
+		"accountID":                   t.pairRecords.selfId.Identifier,
+		"altIRK":                      []byte{0x5e, 0xca, 0x81, 0x91, 0x92, 0x02, 0x82, 0x00, 0x11, 0x22, 0x33, 0x44, 0xbb, 0xf2, 0x4a, 0xc8},
+		"btAddr":                      "FF:DD:99:66:BB:AA",
+		"mac":                         []byte{0xff, 0x44, 0x88, 0x66, 0x33, 0x99},
+		"model":                       "go-ios",
+		"name":                        "host-name",
+		"remotepairing_serial_number": "remote-serial",
+	})
+
+	deviceInfoTlv := newTlvBuffer()
+	deviceInfoTlv.writeData(typeSignature, signature)
```

don'T swallow the errors please

---

As mentioned in the other comment, there will never be an error as `tlvBuffer` uses a `bytes.Buffer` internally